### PR TITLE
Pyflyte meta inputs

### PR DESCRIPTION
**Pull request by kumare3**
# TL;DR
- pyflyte run now shows progress and the entire codebase has been simplified and refactored. 
- Core data string transformation logic has been simplfied and moved to a central place
- This can be extended to support local execution of human in the loop workflows

Screenshot 2023-09-06 at 9 44 15 PM

Also now support all LP options
```
pyflyte --verbose run --remote --env MY_ENV1=VAL1 --env MY_ENV2=VAL2 --label label1=value1 --annotation annotation1=value1 --service-account default --tags my-tag --raw-output-data-prefix s3://.../test-override --max-parallelism 30 hello.py wf
```

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Follow-up issue
Add all options in launchplan to `pyflyte run`

### CODE REVIEW

File: flytekit/clis/sdk_in_container/run.py [hunk 1]
```diff
@@ -1,619 +1,284 @@
...
             default=False,
+            show_default=True,
             help="Whether to overwrite the cache if it already exists",
-        ),
+        )
+    )
+    envvars: typing.Dict[str, str] = make_field(
         click.Option(
-            param_decls=["--envs", "envs"],
+            param_decls=["--envvars", "--env"],
```
Comment (eapolinario) [inline flytekit/clis/sdk_in_container/run.py:151]: ❤️ 

This is a breaking change and will be called out in changelog.

**Review by eapolinario — APPROVED** [review]
